### PR TITLE
Feature/task 313 don't run validators from inactive concepts

### LINKED ISSUE #313: Don't run validators from concepts which are not enabled in a project
We are currently running all registered validators regardless of which concepts are enabled. We should change it to only run concept validators that come from active concepts.

### PR DESCRIPTION
**Pull request by andriikovalov-dlr**
Closes #313

### CODE REVIEW

File: de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/VirSatValidatorBuilder.java [hunk 1]
```diff
@@ -159,34 +153,38 @@ protected void fullBuild(IProgressMonitor monitor, Set<IStructuralElementInstanc
...
-				Object validator = childConfigElement.createExecutableExtension("class");
-				if (validator instanceof IStructuralElementInstanceValidator) {
-					IStructuralElementInstanceValidator seiValidator = (IStructuralElementInstanceValidator) validator;
-					seiValidators.add(seiValidator);
-				} else if (validator instanceof IRepositoryValidator) {
-					IRepositoryValidator repoValidator = (IRepositoryValidator) validator;
-					repoValidators.add(repoValidator);
+			for (IConfigurationElement validatorConfigElement : configElement.getChildren()) {
+				String contributor = validatorConfigElement.getContributor().getName();
+				boolean conceptSpecific = contributor.startsWith("de.dlr.sc.virsat.model.extension");
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/VirSatValidatorBuilder.java]: Maybe we should extract the string into a constant with a descriptive name.

File: de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/VirSatValidatorBuilder.java [hunk 2]
```diff
@@ -152,41 +149,43 @@ protected void fullBuild(IProgressMonitor monitor, Set<IStructuralElementInstanc
...
-				if (validator instanceof IStructuralElementInstanceValidator) {
-					IStructuralElementInstanceValidator seiValidator = (IStructuralElementInstanceValidator) validator;
-					seiValidators.add(seiValidator);
-				} else if (validator instanceof IRepositoryValidator) {
-					IRepositoryValidator repoValidator = (IRepositoryValidator) validator;
-					repoValidators.add(repoValidator);
+			for (IConfigurationElement validatorConfigElement : configElement.getChildren()) {
+				String contributor = validatorConfigElement.getContributor().getName();
+				boolean conceptSpecific = contributor.startsWith(CONCEPT_BUNDLE_PREFIX);
+				if (!conceptSpecific || activeConceptIds.contains(contributor)) {
```
Comment (PhilMFischer) [inline de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/VirSatValidatorBuilder.java]: @andriikovalov-dlr Hey Andrii, I think we are introducing a lot of levels of nesting. Do you think we can reduce them somehow, or is it betetr to keep it like this?

Comment (andriikovalov-dlr) [reply de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/VirSatValidatorBuilder.java]: Yes, one level which we can remove in the future comes from the fact that our extension provides a collection of validators and not just one validator (hence nested for loop). I created a separate ticket for that #315.
On the cosmetic side, we can move validator creation (the `try` block) in a separate method.

Comment (andriikovalov-dlr) [reply de.dlr.sc.virsat.build/src/de/dlr/sc/virsat/build/validator/VirSatValidatorBuilder.java]: Extracted validator creation

**Pull request merged**